File: BHCVRP_Java_Version/src/cujae/inf/citi/om/exceptions/BusStopCapacityException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cujae.inf.citi.om.exceptions;

/**
 *
 * @author nanda
 */

public class BusStopCapacityException extends Exception{
    private int capacityBusStop;
    
    public BusStopCapacityException(int capacityBusStop){
        this.capacityBusStop = capacityBusStop;
    
    }
    
    public int getCapacityBusStop(){
        return capacityBusStop;
    }
    
    public BusStopCapacityException(String message){   
        super(message);
    }
    
}

File: _BHCVRP_Python_Version/cujae_inf_citi_om/exceptions/BusStopCapacityException.py
class BusStopCapacityException(Exception):
    def __init__(self, capacity_bus_stop=None):
        if capacity_bus_stop is not None:
            self.request = capacity_bus_stop

    def __init__(self, message=None):
        if message is not None:
            super().__init__(message)

    def get_capacity_bus_stop(self):
        return getattr(self, "capacity_bus_stop", None)